manifests/init.pp

```diff
@@ -1,24 +1,25 @@
-# == Class: openssl
+# @summary Installs openssl and ensures bundled certificate list is world readable
 #
-# Installs openssl and ensures bundled certificate list is world readable.
+# @param package_name
+#   openssl package name
 #
-# === Parameters
-#  [*package_name*]             openssl package name
-#  [*package_ensure*]           openssl package ensure
-#  [*ca_certificates_ensure*]   ca-certificates package ensure
+# @package_ensure
+#   openssl package ensure
 #
-# === Example
+# @ca_certificates_ensure
+#   ca-certificates package ensure
 #
-#   class { '::openssl':
+# @example basic usage
+#   class { 'openssl':
 #     package_name           => 'openssl-othername',
 #     package_ensure         => latest,
 #     ca_certificates_ensure => latest,
 #   }
 #
 class openssl (
-  Optional[String[1]]       $package_name           = undef,
-  Enum['present', 'absent'] $package_ensure         = present,
-  Enum['present', 'absent'] $ca_certificates_ensure = present,
-){
-  contain '::openssl::packages'
+  Optional[String[1]] $package_name = undef,
+  String[1] $package_ensure = installed,
+  String[1] $ca_certificates_ensure = installed,
```

Comment:
This is to allow `ensure => '1.2.3'` I suppose?

Reply:
Yep, per the docs: https://puppet.com/docs/puppet/7/types/package.html#package-attribute-ensure